Linux/Sources/AfkListener.cpp

```diff
@@ -6,15 +6,40 @@
 #include <spdlog/spdlog.h>
 #include <FocusAfkEventPayload.pb.h>
 #include <FocusSerializer.hpp>
+#include <iostream>
 
 void AfkListener::Run(int triggerAfkInSecond) {
     _triggerAfkInSecond = triggerAfkInSecond;
-    _eventListener = std::make_unique<std::thread>(std::bind(&AfkListener::EventListener, this));
+    setlocale(LC_ALL, "");
+    _display = XOpenDisplay(NULL);
+    if (_display == NULL) {
+        spdlog::get("logger")->error("Failed to connect to X Server");
+    }
+    if (_display != NULL) {
+        _eventListener = std::make_unique<std::thread>(std::bind(&AfkListener::EventListener, this));
+    }
 }
 
 void AfkListener::EventListener() {
-    //TODO: Implementing ContextAgent
-    std::this_thread::sleep_until(std::chrono::system_clock::now() + std::chrono::hours((std::numeric_limits<int>::max)()));
+    bool afk = false;
+    unsigned long lastInputSince = 0;
+    XScreenSaverInfo *info = XScreenSaverAllocInfo();
+
+    while (true) {
+        XScreenSaverQueryInfo(_display, DefaultRootWindow(_display), info);
+        lastInputSince = info->idle / 1000;
+        if (lastInputSince < _triggerAfkInSecond) {
+            afk = false;
+        } else {
+            if (!afk) {
+                spdlog::get("console")->info("AFK since {} seconds", _triggerAfkInSecond);
```

Comment:
Again, you should try to put this in a shared_ptr as per the docs.
```
Note: spdlog::get might slow your code down since it locks a mutex, so use with caution. It is advisable to save the shared_ptr<spdlog::logger> returned and use it directly, at least in code hot paths.
```

Reply:
https://github.com/gabime/spdlog/wiki/2.-Creating-loggers#accessing-loggers-using-spdlogget